extendr-api/tests/extendr_module_tests.rs

```diff
@@ -0,0 +1,28 @@
+//! Tests for [`extendr_module`] procedural macro.
+//!
+mod root {
```

Comment:
Does this test guarantee that we solved the problem correctly? Can we verify the same behaviour in the integration tests, where we already use different modules (but not necessarily nested ones)?

Reply:
I've expanded the test to include an adjacent module.
I don't know what else to do to prove that this is a solution.
The test fails on what was there before.
No other integration test have `extendr_module` for that matter.

Reply:
I was suggesting putting this (or similar code) into `extendrtests` -- an R integration test package. There, if you can safely call one of the functions of an inner module, this guarantees we get correct metadata.

Reply:
But I see you wrote R integration tests.